composite/proof-of-reserves/src/balance.ts

```diff
@@ -18,19 +22,13 @@ export const getBitcoinIndexer = (): BitcoinIndexer | undefined => {
 
 export const getImpl = (options: BitcoinIndexerOptions): Execute => {
   const prefix = options.type?.toUpperCase()
-  switch (options.type) {
-    case BitcoinIndexer.BlockchainCom:
-      return (data) => {
-        const config = blockchainCom.getConfig(prefix)
-        return blockchainCom.execute(data, config)
-      }
+  const impl = options.type && implLookup[options.type]
+  if (impl)
```

Comment:
It's preferable not to nest code if possible:

```javascript
if(!impl) throw Error(`Unknown balance adapter type: ${options.type}`)
return ...
```

Reply:
Yeah guard clauses look way cleaner, good suggestion. Changed.